torch/jit/_script.py

```diff
@@ -1251,12 +1256,12 @@ def _script_impl(
 
 
 def script(
-    obj,
-    optimize=None,
-    _frames_up=0,
-    _rcb=None,
+    obj: Any,
+    optimize: None = None,
+    _frames_up: int = 0,
+    _rcb: Callable[[Any], Any] | None = None,
```

Comment:
This is definitely not typed correctly. Should take some overloads from torch compile to preserve the callable function args if possible and if input is callable.